src/OSPSuite.Core/Domain/UnitSystem/DimensionFactory.cs

```diff
@@ -111,12 +112,7 @@ public IDimension GetOrAddRHSDimensionFor(IDimension dimension)
 
       public IDimension DimensionForUnit(string unitName)
       {
-         var matches = Dimensions.Where(x => x.SupportsUnit(unitName, ignoreCase: true)).ToList();
-         if (!matches.Any())
-            return null;
-
-         //Try to find the first one that matches EXACTLY 
-         return matches.FirstOrDefault(x => x.SupportsUnit(unitName, ignoreCase: false)) ?? matches.First();
+         return Dimensions.DimensionForUnit(unitName);
```

Comment:
@msevestre I moved this code to the newly created extension method to avoid code duplication, since the same function is used in columnInfo.